### ISSUE TITLE
Documentation search can be confusing

### ISSUE THREAD

**Issue by szhorvat** [igraph/python-igraph#611]
Here I'm noting down some potential barriers to using the docs and the doc search, that we discovered with @iosonofabio a while ago.

First, note that the search box on https://igraph.readthedocs.io/ does not search the API.  In order to search the API, one must go to https://igraph.readthedocs.io/en/0.10.2/api/index.html  There, we get live results while typing into the search box.

Suppose we search for betweenness, and I type "betw".  This is what I see:

<img width="816" alt="image" src="https://user-images.githubusercontent.com/1212871/201640245-6f568d0c-712d-4678-baff-8fef76e3096c.png">

Notice that it says, _No results matches "betw"_  At this point, some people will assume that if there are no result for _betw_, there won't be results for _betweenness_ either and stop typing. It turns out that one must type at least _between_ to start getting result.

<img width="827" alt="image" src="https://user-images.githubusercontent.com/1212871/201640504-777728e5-ed02-4984-b11b-e0476dd59508.png">

This is not at all obvious though.

**Comment by ntamas** [comment]
> First, note that the search box on https://igraph.readthedocs.io/ does not search the API. 

That's an unfortunate side effect of trying to use a PyDoctor theme that matches the default theme of the "main" documentation generated by Sphinx independently of PyDoctor. (Basically, the documentation you see is actually two documentations stitched into one). Note how PyDoctor does the same thing; this is the main readthedocs page of PyDoctor: https://pydoctor.readthedocs.io/en/latest/ -- and this is the API doc: https://pydoctor.readthedocs.io/en/latest/api/index.html

The reason why it's less confusing is because the two documentations actually use _different_ themes so you don't get tricked into thinking that the search bar searches both.

As for the minimum character count needed to trigger the search operation, I'll look into this and check whether there's a way to lower the minimum character count for Lunr.

**Comment by ntamas** [comment]
This seems like it's not a character count issue but something else (maybe a similarity threshold)? Searching for `close` yields results much earlier than `between`.

**Comment by ntamas** [comment]
Also note that searching for `bet*` switches Lunr from term search mode to substring search mode and then you start getting results much earlier.

**Comment by szhorvat** [comment]
A particularly bad case is typing `transitivity` which gives zero results despite the existence of `transitivity_avglocal_undirected()`, [transitivity_undirected()](https://python.igraph.org/en/stable/api/igraph.GraphBase.html#transitivity_undirected), [transitivity_local_undirected()](https://python.igraph.org/en/stable/api/igraph.GraphBase.html#transitivity_local_undirected)

**Comment by ntamas** [comment]
I have been looking into this. We would need to patch `search.js` in the documentation generated by PyDoctor to make the behaviour more intuitive, and any patch that we come up with would be fragile - if PyDoctor changes the source of `search.js`, the patch would not apply cleanly any more. I have noticed a similar issue in PyDoctor's issue tracker so I added a comment there in the hope of getting the developers' attention so maybe we could implement an opt-in tweak to the behaviour of the Lunr engine in the next release.
